synapse/storage/databases/main/cache.py

```diff
@@ -168,6 +168,7 @@ def _invalidate_caches_for_event(
         backfilled,
     ):
         self._invalidate_get_event_cache(event_id)
+        self.have_seen_event.prefill((event_id,), True)
```

Comment:
we know what the value would be, so we can prefill - but would it be better just to `.invalidate()` ?